### PR TITLE
Add sample method to IndexingServiceClient

### PR DESCRIPTION
**Pull request by jon-wei**
This PR adds a `sample` method to `IndexingServiceClient` which allows the caller to access the sample API on the overlord. To support this the payload and response classes for the API call (`SamplerSpec` and `SamplerResponse`) have been moved to `server` from the `indexing-service` module.

This PR has:
- [x] been self-reviewed.
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/licenses.yaml)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

**Comment by jon-wei** [comment]
Adding a test now for the new method.

File: server/src/main/java/org/apache/druid/client/indexing/SamplerResponse.java [hunk 1]
```diff
@@ -58,6 +64,27 @@ public int getNumRowsIndexed()
...
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    SamplerResponse that = (SamplerResponse) o;
+    return getNumRowsRead() == that.getNumRowsRead() &&
+           getNumRowsIndexed() == that.getNumRowsIndexed() &&
+           Objects.equals(getData(), that.getData());
```
Comment (abhishekagarwal87) [inline server/src/main/java/org/apache/druid/client/indexing/SamplerResponse.java:79]: this may not do the correct thing always.

Comment (jon-wei) [reply server/src/main/java/org/apache/druid/client/indexing/SamplerResponse.java:79]: What case did you have in mind where this would break down?

Comment (abhishekagarwal87) [reply server/src/main/java/org/apache/druid/client/indexing/SamplerResponse.java:79]: Never mind. I was thinking if, for some `List` implementations, the `equals` may not have been implemented correctly. But that is unlikely to happen.  was also wondering if we care about the order too.

Comment (jon-wei) [reply server/src/main/java/org/apache/druid/client/indexing/SamplerResponse.java:79]: Thanks, IMO responses that have different orders should be considered non-equal

**Review by clintropolis — APPROVED** [review]

**Pull request merged**